test/loader/BasePackageLoader.test.ts

```diff
@@ -0,0 +1,1390 @@
+        __dirname,
+        'fixtures',
+        'StructureDefinition-human-being-logical-model.json'
+      );
+      const pkg = setupLoadPackage(
+        'human-being-logical-model',
+        '1.0.0',
+        'not-loaded',
+        undefined,
+        path.resolve(__dirname, 'fixtures'),
+        humanBeingResourcePath,
+        '2024-05-24T16:27:17-04:00'
+      );
+      packageCacheMock.isPackageInCache.calledWith(pkg.name, pkg.version).mockReturnValue(true);
+      packageCacheMock.getPackagePath
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue(pkg.packageJsonPath);
+      packageCacheMock.getPackageJSONPath
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue(pkg.packageJsonPath);
+      packageCacheMock.getPotentialResourcePaths
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue([pkg.resourcePath]);
+      const humanBeingJSON = fs.readJsonSync(humanBeingResourcePath);
+      packageCacheMock.getResourceAtPath
+        .calledWith(pkg.resourcePath)
+        .mockReturnValue(humanBeingJSON);
+      packageDBMock.getPackageStats
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue({ name: pkg.name, version: pkg.version, resourceCount: 1 });
+
+      const result = await loader.loadPackage('human-being-logical-model', '1.0.0');
+      expect(packageDBMock.savePackageInfo).toHaveBeenCalled();
+      expect(packageDBMock.saveResourceInfo).toHaveBeenCalledWith({
+        resourceType: 'StructureDefinition',
+        id: 'human-being-logical-model',
+        url: 'http://example.org/fhir/locals/StructureDefinition/human-being-logical-model',
+        name: 'Human',
+        version: '1.0.0',
+        sdKind: 'logical',
+        sdDerivation: 'specialization',
+        sdType: 'http://example.org/fhir/locals/StructureDefinition/human-being-logical-model',
+        sdBaseDefinition: 'http://hl7.org/fhir/StructureDefinition/Base',
+        sdAbstract: false,
+        sdCharacteristics: ['can-be-target'],
+        sdFlavor: 'Logical',
+        packageName: 'human-being-logical-model',
+        packageVersion: '1.0.0',
+        resourcePath: humanBeingResourcePath
+      });
+      expect(result).toBe(LoadStatus.LOADED);
+    });
+
+    it('should use LOCAL package paths to find package', async () => {
+      const humanBeingResourcePath = path.resolve(
+        __dirname,
+        'fixtures',
+        'StructureDefinition-human-being-logical-model.json'
+      );
+      const pkg = setupLoadPackage(
+        'LOCAL',
+        'LOCAL',
+        'not-loaded',
+        undefined,
+        path.resolve(__dirname, 'fixtures'),
+        humanBeingResourcePath,
+        '2024-05-24T16:27:17-04:00'
+      );
+      packageCacheMock.isPackageInCache.calledWith(pkg.name, pkg.version).mockReturnValue(true);
+      packageCacheMock.getPackagePath
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue(pkg.packageJsonPath);
```

Comment:
Is the package path supposed to be the same as the `packageJSONPath`? I'm assuming no.

Reply:
I see some others like this too...

Reply:
You are correct: Those should be using `packageBasePath`. I'll clean those up.